### DISCUSSION TITLE
Chose a default bucket to always load

### DISCUSSION THREAD

**Discussion by marouamghar** [alexjustesen/speedtest-tracker#536] [Q&A] [answered]
Whenever I go to the dashboard, I need to re-select the proper bucket, because it just picks the first one in alphabetical order. It can save the older choice so it can go back to it, or pick the default bucket from influxdb settings in the data source section,

**Comment by alexjustesen** [comment]
Not sure I follow, can you share a screenshot on what you mean?

**Comment by marouamghar** [comment]
Sure
![Screenshot from 2023-04-09 08-28-08](https://user-images.githubusercontent.com/78270663/230772534-c7fb773e-6863-4568-bdcb-27be1d32753e.png)
As you can see, when first entering the dashboard, it picks the bucket _monitoring, since it comes first in an alphabetical order.
![Screenshot from 2023-04-09 08-28-18](https://user-images.githubusercontent.com/78270663/230772549-f2163360-6fcf-41d1-9c95-f19d1b5ea0a3.png)
Showing the default order.
![Screenshot from 2023-04-09 08-28-49](https://user-images.githubusercontent.com/78270663/230772551-03a8ff9c-1ca3-4707-ba1e-e561079637b3.png)
I then need to manually switch everytime to default-bucket to get the correct data
![Screenshot from 2023-04-09 08-28-27](https://user-images.githubusercontent.com/78270663/230772550-bb3fd79d-99cc-4bf3-959a-e0c4b9d94b26.png)
The default bucket set up for influxdb is indeed default-bucket

It would be ggod if it can either remember the selected bucket, or just pick the default one from the influxdb config

**Comment by alexdelprete** [comment]
I only have 1 bucket. I don't know what two other buckets you see come from. Where did you get the grafana template?

![image](https://user-images.githubusercontent.com/7027842/230782090-07b78081-5d16-4862-b40c-7e00312d77a4.png)

**Comment by marouamghar** [comment]
i mean, that depends on your influxdb setup. Do you have other buckets on your db that don't appear on the dashboard? If you have only 1 bucket in the influxdb, it makes sense to have only one here.

**Comment by alexdelprete** [comment]
I have a lot of buckets, one for each service. And each bucket has its own token. When you create the datasource for speedtest in grafana, and you specify the token specific for the speedtest bucket, all the others are filtered because that token has rights only for that specific bucket.

So how did you configure your bucket and also your grafana datasource? Here's mine: you can see that when I click Save & Test, only 1 bucket is found because the token I used is mapped to that specific bucket.

![image](https://user-images.githubusercontent.com/7027842/230794119-7c8de76a-25da-474c-9bea-1617dd706c08.png)

**Comment by alexdelprete** [comment]
Also, when you select the bucket from the dashboard, have you tried saving the dashboard? I think it should also save the bucket selection.

![image](https://user-images.githubusercontent.com/7027842/230794188-9b02575c-d1b6-4143-ab27-737812d1934a.png)

**Comment by alexjustesen** [comment]
Converting this to a discussion as it's not directly related to the app.

**Comment by marouamghar** [comment]
The token I use get information for all the 3 buckets, which is the correct behaviour. Now, I am pretty sure if I use a more restricted token to get only the bucket I want for speedtest, I will get only 1 bucket here, and thus it will show only 1 bucket in the dashboard. But that would only be a patch over a wound, since the db can have multiple buckets fetched by one token. In the dashboard we should still be able to select one default bucket to show, or at least save the last bucket shown.
And yes, I did try that save button for the dashboard, it did revert back to the first alphabetical bucket available.

**Reply by alexdelprete** [reply, marked as answer]
> The token I use get information for all the 3 buckets, which is the correct behaviour.

That's not the correct way, from a best practice point of view. But you can do as you want with your data. :)

Anyway, the dashboard uses a function to populate the variable for the bucket list. You can change that in a fixed bucket. Go to edit dashboard, then in variables section, and click on the bucket variable:

![image](https://github.com/alexjustesen/speedtest-tracker/assets/7027842/d15d036a-8312-4c7c-926f-dbdfcc5b3c2d)

In the edit window, you can either modify the Sort to reverse it, so your wanted bucket always comes first:

![image](https://github.com/alexjustesen/speedtest-tracker/assets/7027842/ecfb5c6f-a5aa-4919-80e1-99b5588b12bc)

Or, you change the bucket variable type, from query to custom so you can specify the bucket manually, like in the following screenshot:

![image](https://github.com/alexjustesen/speedtest-tracker/assets/7027842/6c980fec-10d4-4acd-9f64-78ea75a6abfd)

And then save the dashboard.

**Reply by marouamghar** [reply]
> The token I use get information for all the 3 buckets, which is the correct behaviour.

 I meant the dashboard is behaving correctly by getting all available buckets, not that using one token to get all info is the correct behavior. Small confusion there.
I just got time to go back to this, and it works. Thanks for your help !
